master/internal/api_experiment.go

```diff
@@ -124,6 +124,19 @@ func (a *apiServer) GetExperiments(
 	return resp, a.paginate(&resp.Pagination, &resp.Experiments, req.Offset, req.Limit)
 }
 
+func (a *apiServer) GetExperimentLabels(_ context.Context,
+	req *apiv1.GetExperimentLabelsRequest) (*apiv1.GetExperimentLabelsResponse, error) {
+	resp := &apiv1.GetExperimentLabelsResponse{}
+
+	var err error
+	resp.Labels, err = a.m.db.ExperimentLabels()
+	if err != nil {
+		return nil, err
```

Comment:
wdy think about wrapping the error to add more context on what' happened (higher level than a specific database error)

we could also use grpc/status to set specific status codes which also map to preset http error codes although here it'd would probably be a 500 either way.

Reply:
I don't think there's any context I can add at this level, other than which API call the error occurred in. But as we're returning it, not logging it, they already know that. I can add context to the JSON parsing one to make it easy to find the root cause if it ever comes up again, though, and I've checked that it does yield a 500 when it triggers.